src/plugins/home/public/application/components/welcome.tsx

```diff
@@ -141,19 +145,39 @@ export class Welcome extends React.Component<Props> {
   };
 
   render() {
-    const { urlBasePath, telemetry } = this.props;
+    const { urlBasePath, telemetry, branding } = this.props;
     return (
       <EuiPortal>
         <div className="homWelcome">
           <header className="homWelcome__header">
             <div className="homWelcome__content eui-textCenter">
               <EuiSpacer size="xl" />
-              <span className="homWelcome__logo">
-                <EuiIcon type={OpenSearchMarkCentered} size="original" />
-              </span>
-              <EuiTitle size="l" className="homWelcome__title">
+              {branding.smallLogoUrl === undefined ? (
```

Comment:
sorry for not catching this the first pass, but i think we can pull this logic out and pass a function for example checkout out renderTelemetryEnabledOrDisabledText() function in this file.

Reply:
Agreed. Wrote a similar function called renderBrandingEnabledOrDisabledLogo() function.